glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -646,4 +646,62 @@ class GleanTest {
 
         Dispatchers.API.overflowCount = 0
     }
+
+    @Test
+    fun `sending deletion ping if disabled outside of run`() {
+        val server = getMockWebServer()
+        resetGlean(
+            context,
+            Glean.configuration.copy(
+                serverEndpoint = "http://" + server.hostName + ":" + server.port,
+                logPings = true
+            ),
+            uploadEnabled = true
+        )
+
+        resetGlean(
+            context,
+            Glean.configuration.copy(
+                serverEndpoint = "http://" + server.hostName + ":" + server.port,
+                logPings = true
+            ),
+            uploadEnabled = false,
+            clearStores = false
+        )
+
+        // Now trigger it to upload
+        triggerWorkManager(context, DeletionPingUploadWorker.PING_WORKER_TAG)
+
+        val request = server.takeRequest(20L, TimeUnit.SECONDS)
+        val docType = request.path.split("/")[3]
+        assertEquals("deletion-request", docType)
+    }
+
+    @Test
+    fun `no sending of deletion ping if unchanged outside of run`() {
+        val server = getMockWebServer()
+        resetGlean(
```

Comment:
Ideally we'd also want to check that we're not mistakenly resetting the client id for extra safety. Unless that's done in some existing test already

Reply:
It might be, but it's not a bad idea to do it here as well.